src/features/ProductPage/components/ProductListing.jsx

```diff
@@ -0,0 +1,102 @@
+import React from 'react';
+import {useSelector, useDispatch} from 'react-redux';
+import {setProducts, addProducts} from '../slices/productSlice';
+import {fetchProducts} from '../../../common/utils/api';
+import {useState, useEffect} from 'react';
+import {PRODUCT_LISTING_LIMIT} from '../../../constants';
+import Product from './Product';
+import {BiLoaderCircle} from 'react-icons/bi';
+
+const ProductListing = () => {
+  const dispatch = useDispatch();
+  const products = useSelector((state) => state.products.products);
+  const filters = useSelector((state) => state.filters.filters);
```

Comment:
Your reducer functions should be exported from the slice declaration file itself, as defining then this way will lead to unnecessary garbage-collection and re-allocation of browser-retained RAM for the anonymous functions.

please check reference here:
https://github.com/browserstack/frontend/blob/04bd0b849492b7cfac75d9823bf9f14693190231/apps/mcspt/src/features/Home/slices/newPerformanceSessionSlice.js

Reply:
Skipping for now.

Point noted.

Reply:
Cant approve without these changes.